test/helpers/vmware/unit_test_context.go

```diff
@@ -49,20 +49,20 @@ type UnitTestContextForController struct {
 // NewUnitTestContextForController returns a new UnitTestContextForController
 // with an optional prototype cluster for unit testing controllers that do not
 // invoke the VSphereCluster spec controller.
-func NewUnitTestContextForController( /*newReconcilerFn NewReconcilerFunc, */ namespace string, vSphereCluster *vmwarev1.VSphereCluster,
+func NewUnitTestContextForController( /*newReconcilerFn NewReconcilerFunc, */ ctx context.Context, namespace string, vSphereCluster *vmwarev1.VSphereCluster,
```

Comment:
Let's also remove the commented out parameter here. It got commented out in https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1610 without any bigger note.


```suggestion
func NewUnitTestContextForController(ctx context.Context, namespace string, vSphereCluster *vmwarev1.VSphereCluster,
```